code/datums/diseases/advance/symptoms/heal.dm

```diff
@@ -1,161 +1,289 @@
 
-	DNA Restoration
+	if(!parts.len)
+		return
 
-	Not well hidden.
-	Lowers resistance minorly.
-	Does not affect stage speed.
-	Decreases transmittablity greatly.
-	Very high level.
+	if(prob(5))
+		to_chat(M, "<span class='notice'>You feel yourself absorbing the water around you to soothe your damaged skin.</span>")
 
-Bonus
-	Heals brain damage, treats radiation, cleans SE of non-power mutations.
+	for(var/obj/item/organ/external/L in parts)
+		if(L.heal_damage(heal_amt/parts.len * 0.5, heal_amt/parts.len, null, TRUE))
+			M.UpdateDamageIcon()
 
-//////////////////////////////////////
-*/
+	return 1
 
-/datum/symptom/heal/dna
+/datum/symptom/heal/water/passive_message_condition(mob/living/M)
+	if(M.getBruteLoss() || M.getFireLoss())
+		return TRUE
+	return FALSE
 
-	name = "Deoxyribonucleic Acid Restoration"
+/datum/symptom/heal/radiation
+	name = "Radioactive Resonance"
+	desc = "The virus uses radiation to fix damage through dna mutations."
 	stealth = -1
-	resistance = -1
-	stage_speed = 0
+	resistance = -2
+	stage_speed = 2
 	transmittable = -3
-	level = 5
-
-/datum/symptom/heal/dna/Heal(mob/living/carbon/M, datum/disease/advance/A)
-	var/amt_healed = (sqrtor0(20+A.totalStageSpeed()*(3+rand())))-(sqrtor0(16+A.totalStealth()*rand()))
-	M.adjustBrainLoss(-amt_healed)
-	//Non-power mutations, excluding race, so the virus does not force monkey -> human transformations.
-	var/list/unclean_mutations = (not_good_mutations|bad_mutations) - mutations_list[RACEMUT]
-	M.dna.remove_mutation_group(unclean_mutations)
-	M.radiation = max(M.radiation - 3, 0)
+	level = 6
+	symptom_delay_min = 1
+	symptom_delay_max = 1
+	passive_message = "<span class='notice'>Your skin glows faintly for a moment.</span>"
+	var/cellular_damage = FALSE
+	threshold_desc = "<b>Transmission 6:</b> Additionally heals cellular damage.<br>\
+					  <b>Resistance 7:</b> Increases healing speed."
+
+/datum/symptom/heal/radiation/Start(datum/disease/advance/A)
+	if(!..())
+		return
+	if(A.properties["resistance"] >= 7)
+		power = 2
+	if(A.properties["transmittable"] >= 6)
+		cellular_damage = TRUE
+
+/datum/symptom/heal/radiation/CanHeal(datum/disease/advance/A)
+	var/mob/living/M = A.affected_mob
+	switch(M.radiation)
+		if(0)
+			return FALSE
+		if(1 to 25)
+			return 0.25
+		if(25 to 50)
+			return 0.5
+		if(50 to 75)
+			return 0.75
+		if(75 to 100)
+			return 1
+		else
+			return 1.5
+
+/datum/symptom/heal/radiation/Heal(mob/living/carbon/human/M, datum/disease/advance/A, actual_power)
+	var/heal_amt = actual_power
+
+	if(cellular_damage)
+		M.adjustCloneLoss(-heal_amt * 0.5)
+
+	M.adjustToxLoss(-(2 * heal_amt))
+
+	var/list/parts = M.get_damaged_organs(1, 1)
+
+	if(!parts.len)
+		return
+
+	if(prob(4))
+		to_chat(M, "<span class='notice'>Your skin glows faintly, and you feel your wounds mending themselves.</span>")
+
+	for(var/obj/item/organ/external/L in parts)
+		if(L.heal_damage(heal_amt/parts.len, heal_amt/parts.len, null, 1))
```

Comment:
LAZYLEN and spacing

Reply:
going to keep this the same because using LAZYLEN kind of hurts readability here imo